src/java/org/apache/cassandra/cql3/constraints/UnaryFunctionColumnConstraint.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.constraints;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.cql3.CqlBuilder;
+import org.apache.cassandra.db.TypeSizes;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.tcm.serialization.MetadataSerializer;
+import org.apache.cassandra.tcm.serialization.Version;
+import org.apache.cassandra.utils.LocalizeString;
+
+import static org.apache.cassandra.cql3.constraints.ColumnConstraint.ConstraintType.UNARY_FUNCTION;
+
+public class UnaryFunctionColumnConstraint implements ColumnConstraint<UnaryFunctionColumnConstraint>
+{
+    public static final Serializer serializer = new Serializer();
+
+    public final ConstraintFunction function;
+    public final ColumnIdentifier columnName;
+
+    public final static class Raw
+    {
+        public final ConstraintFunction function;
+        public final ColumnIdentifier columnName;
+
+        public Raw(ColumnIdentifier functionName, ColumnIdentifier columnName)
+        {
+            this.columnName = columnName;
+            function = createConstraintFunction(functionName.toCQLString(), columnName);
+        }
+
+        public UnaryFunctionColumnConstraint prepare()
+        {
+            return new UnaryFunctionColumnConstraint(function, columnName);
+        }
+    }
+
+    private enum Functions
+    {
+        NOT_NULL(NotNullConstraint::new);
+
+        private final Function<ColumnIdentifier, ConstraintFunction> functionCreator;
+
+        Functions(Function<ColumnIdentifier, ConstraintFunction> functionCreator)
+        {
+            this.functionCreator = functionCreator;
+        }
+    }
+
+    private static ConstraintFunction createConstraintFunction(String functionName, ColumnIdentifier columnName)
```

Comment:
This, and some other methods are identical to the ones in `FunctionColumnConstraint`. Should we share that logic with an abstract class? I mean, it is a "small" repetition, but still somehow annoys me to see it.

Reply:
It was trying to do that but I was not sure how to abstract `Functions` enum. That would need to be passed as a parameter and I was stuck doing that. Feel free to try it.